examples/material_gallery/lib/gallery/demo.dart

```diff
@@ -211,25 +212,58 @@ class _FormattedCodeState extends State<FormattedCode> {
   void didUpdateConfig(FormattedCode oldConfig) {
     super.didUpdateConfig(oldConfig);
 
-    if (oldConfig.code != config.code)
+    if (oldConfig.exampleCode != config.exampleCode)
       _formatText();
   }
 
   void _formatText() {
     _formattedText = new TextSpan(
       style: new TextStyle(fontFamily: 'monospace', fontSize: 10.0),
-      children: <TextSpan>[new DartSyntaxHighlighter().format(config.code)]
+      children: <TextSpan>[new DartSyntaxHighlighter().format(config.exampleCode)]
     );
   }
 }
 
-class FullScreenCodeDialog extends StatelessWidget {
-  FullScreenCodeDialog({ this.code });
+class FullScreenCodeDialog extends StatefulWidget {
+  FullScreenCodeDialog({ this.exampleCodeTag });
+
+  final String exampleCodeTag;
+
+  @override
+  FullScreenCodeDialogState createState() => new FullScreenCodeDialogState();
+}
+
+class FullScreenCodeDialogState extends State<FullScreenCodeDialog> {
 
-  final String code;
+  String _exampleCode;
+
+  @override
+  void initState() {
+    super.initState();
+
+    getExampleCode(config.exampleCodeTag, context).then((String code) {
```

Comment:
rather than passing the context, I would pass the asset bundle. That way the code that does the parsing doesn't have to know anything about Widgets.